File: solutions/2026-01-17/solution.java
// Valid Anagram
// Difficulty: Medium
// URL: https://leetcode.com/problems/valid-anagram/

class Solution {
  public boolean backspaceCompare(String s, String t) {
    return buildString(s).equals(buildString(t));
  }

  private String buildString(String s) {
    StringBuilder sb = new StringBuilder();
    for (char c : s.toCharArray()) {
      if (c != '#') {
        sb.append(c);
      } else {
        if (sb.length() > 0) {
          sb.deleteCharAt(sb.length() - 1);
        }
      }
    }
    return sb.toString();
  }
}

File: solutions/2026-01-17/solution.py
# Valid Anagram
# Difficulty: Medium
# URL: https://leetcode.com/problems/valid-anagram/

def isBackspaceEqual(s: str, t: str) -> bool:
  def getStr(s: str) -> str:
    res = ''
    for c in s:
      if c != '#':
        res += c
      else:
        res = res[:-1]
    return res
  return getStr(s) == getStr(t)